saleor/checkout/models.py

```diff
@@ -299,3 +299,9 @@ def __setstate__(self, data):
     def is_shipping_required(self) -> bool:
         """Return `True` if the related product variant requires shipping."""
         return self.variant.is_shipping_required()
+
+
+class CheckoutMetadata(ModelWithMetadata):
+    checkout = models.OneToOneField(
+        Checkout, related_name="metadata_storage", on_delete=models.CASCADE
+    )
```

Comment:
I think that you need to add `class Meta(ModelWithMetadata.Meta):` here. Without this, the indexes declared for metadata will not be created.

Reply:
Will they? with just that `makemigrations` is trying to generate those indexes properly, filtering also works fine. Should anything be checked to make sure the indexes are properly created?

Reply:
Hhm, I was sure, that I didn't find the indexes in the migrations, but it looks like it placed there.